src/repo/gitlab.js

```diff
@@ -0,0 +1,88 @@
+const pify = require('pify')
+const request = pify(require('request'))
+
+const getUserInfo = function(username, hostname) {
+  /* eslint-disable complexity */
+  if (!hostname) {
+    hostname = 'https://gitlab.com';
+  }
+
+  return request
+    .get({
+      url: `${hostname}/api/v4/users?username=${username}`,
+      headers: {
+        'User-Agent': 'request',
+      },
+    })
+    .then(res => {
+      const body = JSON.parse(res.body)
+
+      // Gitlab returns an array of users. If it is empty, it means the username provided does not exist
+      if (!body || body.length === 0) {
+        throw new Error(`User ${username} not found`)
+      }
+
+      const user = body[0]
+
+      return {
+        login: user.username,
+        name: user.name || username,
+        avatar_url: user.avatar_url,
+        profile: user.web_url.startsWith('http') ? user.web_url : `http://${user.web_url}`,
+      }
+    })
+}
+
+const getContributors = function(owner, name, hostname) {
+  if (!hostname) {
+    hostname = 'https://gitlab.com';
+  }
+
+  return request
+    .get({
+      url: `${hostname}/api/v4/projects?search=${name}`,
+      headers: {
+        'User-Agent': 'request',
+      },
+    })
+    .then(res => {
+      const projects = JSON.parse(res.body)
+
+      // Gitlab returns an array of users. If it is empty, it means the username provided does not exist
+      if (!projects || projects.length === 0) {
+        throw new Error(`Project ${owner}/${name} not found`)
+      }
+
+      let project = null
+      for (let i = 0; i < projects.length; i++) {
+        if (projects[i].path_with_namespace === `${owner}/${name}`) {
+          project = projects[i]
+          break;
+        }
+      }
+
+      if (!project) {
+        throw new Error(`Project ${owner}/${name} not found`)
+      }
+
+      return request
+        .get({
+          url: `${hostname}/api/v4/projects/${project.id}/repository/contributors`,
+          headers: {
+            'User-Agent': 'request',
+          },
+        })
+        .then(newRes => {
+          const contributors = JSON.parse(newRes.body)
+					if (newRes.statusCode >= 400) {
+		        throw new Error(contributors.message)
+		      }
+          return contributors.map(item => item.name)
```

Comment:
So sad that GitLab doesn't return the username.
I tested by changing my name in the `.allcontributorsrc` file to "Mehdi Acho".

Running the check command gives:
```
Missing contributors in .all-contributorsrc:
    Mehdi Achour
```

It also should have mentioned the unknown contributor found (Acho).
This happens around line 108 in src/cli.js

Reply:
This comment got hidden as lines were changed. Don't forget to check it ;)

Reply:
You're right I had missed it! So basically the error is that it detects the Gitlab user is not in the file, but does not detect that the contributor in the file is not in Gitlab. Is that it?

Reply:
exactly "is not in the project's contributors as returned by Gitlab" to be precise